src/neo/IO/Caching/DataCache.cs

```
@@ -110,7 +110,7 @@ public void Commit()
         /// Delete key from cached Dictionary or search in Internal.
         /// </summary>
         /// <param name="key">Key to be deleted.</param>
-        public void Delete(TKey key)
+        public bool Delete(TKey key)
```

Does this change needs to come in this PR?
Maybe it is better to open a new one first and we can approve it beforehand.

---

It's used in `Burn` method L236

```cs
if (!storages.Delete(tokenKey)) throw new InvalidOperationException("Token doesn't exist");
```

---

But why throw the exception?

---

It's not a valid NFT.

_The parameter tokenid SHOULD be a valid NFT. If not, this method SHOULD throw an exception._  https://github.com/neo-project/proposals/pull/41

---

It means when transferring, so you should throw an exception while transferring.

---

I think we should keep it consistency in other method, like `Burn` method. But if you prefer to remove it, we can adjust it.

---

Let's keep it for a look.